### PR TITLE
Custom Product Attribute changes backend_type value

### PR DESCRIPTION
**Pull request by keyurshah070**
### Fixed Issues (if relevant)

1. magento/magento2#9219: Custom Product Attribute changes 'backend_type' when 'is_user_defined = 1' and get updated/saved in Admin Backend
2. Custom user define product attribute with backend type change while saving from backend.

### Manual testing scenarios

1. Created a new module with Installer script with which contain attribute create a script with backendtype as varchar and change its property from backend and after saving attribute still, I have backendtype as varchar so I think this is good.

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Save.php [hunk 1]
```diff
@@ -192,11 +192,12 @@ public function execute()
                     $data['frontend_input']
                 );
             }
-
             $data += ['is_filterable' => 0, 'is_filterable_in_search' => 0, 'apply_to' => []];
 
-            if (is_null($model->getIsUserDefined()) || $model->getIsUserDefined() != 0) {
-                $data['backend_type'] = $model->getBackendTypeByInput($data['frontend_input']);
+            if ($model->getIsUserDefined()) {
+                if (empty($model->getIsUserDefined())) {
```
Comment (orlangur) [inline app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Save.php]: For which value of `$model->getIsUserDefined()` will it ever pass into?

Comment (keyurshah070) [reply app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Save.php]: Oh, sorry bad commit ,

**Comment by omiroshnichenko** [comment]
@keyurshah070, please, read comment on https://github.com/magento/magento2/pull/11715#issuecomment-345237460.

**Comment by keyurshah070** [comment]
Hi @omiroshnichenko I resolved the conflict. please check

**Pull request closed without merging**